snapshot/src/main/java/io/zeebe/snapshots/broker/impl/FileBasedSnapshotStoreFactory.java

```diff
@@ -43,7 +38,14 @@ public PersistedSnapshotStore createSnapshotStore(final Path root, final String
     IoUtil.ensureDirectoryExists(snapshotDirectory.toFile(), "Snapshot directory");
     IoUtil.ensureDirectoryExists(pendingDirectory.toFile(), "Pending snapshot directory");
 
-    return new FileBasedSnapshotStore(
-        new SnapshotMetrics(partitionName), snapshotDirectory, pendingDirectory);
+    return partitionSnapshotStores.computeIfAbsent(
+        partitionName,
+        p ->
+            new FileBasedSnapshotStore(
+                new SnapshotMetrics(partitionName), snapshotDirectory, pendingDirectory));
+  }
+
+  public FileBasedSnapshotStore getSnapshotStore(final String partitionName) {
```

Comment:
Please always try to return the interfaces

Reply:
Why was this necessary? To get the same store in the ZeebePartition as in the journal?

Reply:
Is is problematic if we never really recreate the store when we had a leader change?